### ISSUE TITLE
AGWG feedback: Element in sequential focus order has visible focus

### ISSUE THREAD

**Issue by WilcoFiers** [w3c/wcag-act-rules#224]
Source: https://www.w3.org/2002/09/wbs/35422/act-dec-2022/results#xq6

Comments grouped by, and responded to by topic

## One Pixels

> _From Oliver Keim_
> 1 pixel seems not enough. A larger portion is needed.

> _From Gregg Vanderheiden_
> on pixel in a different color does not mean it is visible.
> 1) one pixel is not really visible
> 2) the pixel could be one the same color with a least significant digit incremented by 1 - and it would be a different color but not distinguishable even with a magnifying glass.
> 
> I appreciate the difficulty since "visible" is not defined - (and this shouldn't have passed without a definition of visible) so I am not sure what to do here - but this in effect creates a definition of visible that is not in WCAG and therefore is not allowed any more than a more reasonable ("someone with 20:20 vision can reliably tell the focused items from the non focused items when 20 feet from the screen"

## Automation & Algorithms

> _From Stefan Schnabel_
> Please add examples of machine algorithms how to test this !!!

> _From Jaunita George	:_
> This is also likely going to generate a significant number of false positives.

## Other

> _From Michael Gower_
> I think this must list Focus Order https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html in the requirements mapping, or is there something I'm missing?
> 
> Most implementations of 'return to top' put the user back at the top of the page without showing a focus indicator. Some may find it confusing if the focus went around a non-interactive heading (which is typically where focus goes). That said, in most other circumstances where focus goes on non-UI components, there's usually a visible indicator.
> 
> Focus Visible does not seem to address if it's just for UIC components. This is probably another hole that needs to be sorted out as part of WCAG 3.0.

> _From Detlev Fischer_
> Background: "All Examples in this rule satisfy those success criteria."
> 
> Really all? I guess this should read: "All *Passed* Examples in this rule satisfy those success criteria."

> Andrew Kirkpatrick
> Unicity is an unusual word. How about "WCAG does not require that the focus indicator for each focusable element is unique in appearance." instead of "WCAG has no clear requirement of unicity of the focus indicator for each focusable element"?

**Comment by WilcoFiers** [comment]
## Proposed Task Force Response

Response per topic:

### One pixel

ACT Rules test for failure conditions. The rule isn't saying that a single pixel focus indicator is sufficient to satisfy this success criterion. It is saying that a 0 pixels focus indicator fails the success criterion. The "Accessibility requirements" section describes what the outcome of an ACT Rule means for the Focus Visible success criterion. Failing this rule means the success criterion is not satisfied. Passing the rule means the success criterion requires further testing.

If AGWG wants to suggest a minimum area and minimum contrast for this success criterion, even in the understanding document, this would be incorporated into the rule. Lacking such a description, the only thing we can reasonably conclude is that 0 pixels is insufficient, and anything more than that is left open to interpretation.

We do hope to have an ACT Rule for WCAG 2.2's Focus Appearance success criterion, which may also help address this issue of not having a minimum for what is considered a "visible" focus indicator.

### Automation & Algorithms

ACT Rules are precise descriptions of what the outcome of a test should be. They intentionally do not include algorithms of how to perform a test. This was done because there are often different ways of achieving the same test result. How an testers (tools and humans) do their work is often proprietary information as well.

ACT Rules are not intended to as quality control for tools and test methodologies. The test cases included in the rules are sufficient to gage that an implementation incorporated all the important aspects covered by the rule, but they are not exhaustive enough to ensure tests happen without false positives. This is the responsibility of tool and methodology developers.

Rules themselves can sometimes have false positives. We actively encourage implementors to report any false positives they find in rules so that we can address them. ACT Task Force reviews any open issues before submitting a rule to AG. On this particular rule we have no open issues.

### Other

The editorial issues have been addressed.

On Mike's comment, this rule is limited to elements in sequential focus navigation. This includes all elements that can be focused on using the tab key. Some of those may not be user interface components. The targets of move-to-top links are only included if they are focusable through tab navigation.

**Comment by kengdoj** [comment]
One Pixel: The rule's Expectation says "at least one device pixel" so recommend using that terminology instead of "0":

> It is saying that a 0 pixels focus indicator fails the success criterion.

- It is saying that a focus indicator _less than 1_ pixel fails the success criterion.

> Lacking such a description, the only thing we can reasonably conclude is that 0 pixels is insufficient, and anything more than that is left open to interpretation.
- I'm ok with using 0 here. But for the sake of consistency, this could be "Lacking such a description, the only thing we can reasonably conclude is that _less than 1 pixel_ is insufficient."

Automation & Algorithms: 
> How an testers (tools and humans) do their work is often proprietary information as well.
- first paragraph last sentence, remove "an"

> ACT Rules are not intended to as quality control for tools and test methodologies. 
- 2nd paragraph first sentence, remove "to"


In the [rule's Background](https://www.w3.org/WAI/standards-guidelines/act/rules/oj04fd/proposed/#background), change "E" to lower case:

> and all Examples in this rule avoid it
